### PR TITLE
fix futhark build failed in ghc 8.10.3

### PR DESCRIPTION
**Pull request by fishtreesugar**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
fixes for #67284, cause by `process-extras`'s transitive dependency of `ListLike` (ddssff/listlike#8)

### CODE REVIEW

**Comment by carlocab** [comment]
Thanks for this. Haven't had a chance to take a closer look at the CI failures so I wasn't able to figure out how to fix them yet.

File: Formula/futhark.rb [hunk 1]
```diff
@@ -22,7 +22,7 @@ class Futhark < Formula
 
   def install
     system "cabal", "v2-update"
-    system "cabal", "v2-install", *std_cabal_v2_args
+    system "cabal", "v2-install", *std_cabal_v2_args, "--constraint=bytestring==0.10.10.1"
```
Comment (carlocab) [inline Formula/futhark.rb:25]: Would this need to be updated with version bumps? If so, a comment would be good the explains how.

File: Formula/futhark.rb [hunk 2]
```diff
@@ -5,6 +5,7 @@ class Futhark < Formula
   sha256 "262dd8024c58ebb05964010227bc76907c873f261ab3348ffeab5fa5b1def022"
   license "ISC"
   head "https://github.com/diku-dk/futhark.git"
+  revision 1
```
Comment (carlocab) [inline Formula/futhark.rb:8]:
```suggestion
  revision 1
  head "https://github.com/diku-dk/futhark.git"
```
Also, I don't actually think this needs a revision, as the change you've made is relevant only for the build.

**Comment by carlocab** [comment]
Picked up your commit with minor revisions, and included it here: https://github.com/Homebrew/homebrew-core/pull/67284

Hopefully, it fixes some build failures...

File: Formula/futhark.rb [hunk 3]
```diff
@@ -4,6 +4,7 @@ class Futhark < Formula
   url "https://github.com/diku-dk/futhark/archive/v0.18.4.tar.gz"
   sha256 "262dd8024c58ebb05964010227bc76907c873f261ab3348ffeab5fa5b1def022"
   license "ISC"
+  revision 1
```
Comment (carlocab) [inline Formula/futhark.rb:7]:
```suggestion
```
The revision is not needed.

**Comment by carlocab** [comment]
You can actually close this now. If/when the ghc PR is merged this commit will be recorded as authored by you: https://github.com/Homebrew/homebrew-core/pull/67284/commits/f8c0738d5043b1bfb5a4dc856c3a3f7b66319278

**Comment by fishtreesugar** [comment]
Thanks.

**Pull request closed without merging**